webapi/src/main/scala/org/knora/webapi/routing/v1/ValuesRouteV1.scala

```diff
@@ -422,14 +429,52 @@ object ValuesRouteV1 extends Authenticator {
                         // TODO: how to check if the user has sent multiple files?
                         val nonEmpty: NonEmpty = bodyPartFile.entity
                             .toOption.getOrElse(throw BadRequestException("no binary data submitted in multipart request"))
+                        */
+
+                        val userProfile = getUserProfileV1(requestContext)
+
+                        type Name = String
+
+                        @volatile var receivedFile: Option[File] = None
+
+                        /* get the file data */
+                        val fileMapFuture = formdata.parts.mapAsync(1) { p: Multipart.FormData.BodyPart =>
+                            if (p.name == "file") {
+                                //println(s"part named ${p.name} has file named ${p.filename}")
+                                val filename = p.filename.getOrElse(throw BadRequestException(s"Filename is not given"))
+                                val tmpFile = InputValidation.createTempFile(settings)
+                                val written = p.entity.dataBytes.runWith(FileIO.toPath(tmpFile.toPath))
+                                written.map { written =>
+                                    //println(s"written result: ${written.wasSuccessful}, ${p.filename.get}, ${tmpFile.getAbsolutePath}")
+                                    receivedFile = Some(tmpFile)
+                                    Map(p.name -> FileInfo(p.name, p.filename.get, p.entity.contentType))
+                                }
+                            } else {
+                                Future(Map.empty[Name, FileInfo])
+                            }
+                        }.runFold(Map.empty[Name, FileInfo])((set, value) => set ++ value)
+
+                        val fileMap = Await.result(fileMapFuture, Timeout(10.seconds).duration)
+
 
                         // save file to temporary location
                         // this file will be deleted by Knora once it is not needed anymore
                         // TODO: add a script that cleans files in the tmp location that have a certain age
                         // TODO  (in case they were not deleted by Knora which should not happen -> this has also to be implemented for Sipi for the thumbnails)
+                        /*
                         val sourcePath = InputValidation.saveFileToTmpLocation(settings, nonEmpty.data.toByteArray)
                         val originalFilename = bodyPartFile.filename.getOrElse(throw BadRequestException(s"Filename is not given"))
                         val originalMimeType = nonEmpty.contentType.toString
+                        */
+
+                        // FIXME: Make it safer
```

Comment:
``receivedFile`` is an option. I can do a quick fix with ``receivedFile.getOrElse(throw ExceptionXYZ)``